invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/CommunitiesList.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from "react";
+// import PropTypes from "prop-types";
+import axios from "axios";
+import { withCancel } from "react-invenio-forms";
+import { Container, Item, Button, Placeholder, Message } from "semantic-ui-react";
+import { Image } from "react-invenio-forms";
+import { CommunitiesListModal } from "./CommunitiesListModal/CommunitiesListModal";
+
+export class CommunitiesList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      communities: null,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.getCommunities();
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetchCommunities?.cancel();
+  }
+
+  getCommunities = async () => {
+    this.cancellableFetchCommunities = withCancel(this.fetchRecordCommunities());
+    this.setState({
+      loading: true,
+      error: null,
+    });
+    try {
+      const res = await this.cancellableFetchCommunities.promise;
+      this.setState({
+        communities: res.data?.hits?.hits,
+        loading: false,
+      });
+    } catch (error) {
+      if (error !== "UNMOUNTED") {
+        this.setState({
+          loading: false,
+          error: i18next.t("An error occurred while fetching communities."),
+        });
+      }
+    }
+  };
+
+  fetchRecordCommunities = async () => {
+    // TODO: Update endpoint to be record specific
+    return await axios.get("/api/communities", {
+      headers: {
+        Accept: "application/vnd.inveniordm.v1+json",
+      },
+    });
+  };
+
+  render() {
+    const { communities, loading, error } = this.state;
+    const communityItems = communities?.map((community, index) => {
+      if (index > 2) return null;
+      return (
+        <Item key={community.id}>
+          <Image size="mini" src={community.links.logo} />
+          <Item.Content verticalAlign="middle">
+            <Item.Header
+              as="a"
+              href={`/communities/${community.slug}`}
+              className="ui small header"
+            >
+              {community.metadata.title}
+            </Item.Header>
+          </Item.Content>
+        </Item>
+      );
+    });
+
+    return (
+      <>
+        {(loading && (
+          <Placeholder>
+            <Placeholder.Header image>
+              <Placeholder.Line />
+              <Placeholder.Line />
+            </Placeholder.Header>
+          </Placeholder>
+        )) || <Item.Group>{communityItems}</Item.Group>}
+
+        {!loading && communities?.length > 3 && (
+          <Container align="center" className="mt-10">
+            <CommunitiesListModal
+              trigger={
+                <Button size="tiny" className="transparent">
+                  {i18next.t("View all communities")}
+                </Button>
+              }
+            />
+          </Container>
+        )}
+
+        {error && <Message error>{error}</Message>}
+      </>
+    );
+  }
+}
+
+// CommunitiesList.propTypes = {
```

Comment:
leftovers?

Reply:
I think it is OK already to have them, no need to comment out. Instead of passing recordId as a prop, and then hardcoding the URL as `/api/records/{recordDd}/communities` (which is not ideal solution) you could simply pass the whole endpoint `recordCommunitySearchEndpoint` as a prop and set it in jinja. It will be more flexible to change in the future if done this way